crates/turborepo-scm/src/lib.rs

```diff
@@ -40,43 +40,54 @@ impl Error {
     pub(crate) fn git_error(s: impl Into<String>) -> Self {
         Error::Git(s.into(), Backtrace::capture())
     }
-
-    fn from_git_exit_code(cmd: &str, path: &AbsoluteSystemPath, exit_code: Option<i32>) -> Error {
-        let s = format!(
-            "'{}' in {} exited with status code {}",
-            cmd,
-            path.to_string(),
-            exit_code
-                .map(|code| code.to_string())
-                .unwrap_or("unknown".to_string())
-        );
-        Error::Git(s, Backtrace::capture())
-    }
 }
 
-pub(crate) fn read_git_error<R: Read>(stderr: &mut R) -> Option<Error> {
+fn read_git_error_to_string<R: Read>(stderr: &mut R) -> Option<String> {
     let mut buf = String::new();
     let bytes_read = stderr.read_to_string(&mut buf).ok()?;
     if bytes_read > 0 {
         // something failed with git, report that error
-        Some(Error::git_error(buf))
+        Some(buf)
     } else {
         None
     }
 }
 
-pub(crate) fn wait_for_success<R: Read>(
+pub(crate) fn wait_for_success<R: Read, T>(
     mut child: Child,
     stderr: &mut R,
     command: &str,
     root_path: impl AsRef<AbsoluteSystemPath>,
-) -> Result<(), Error> {
+    parse_result: Result<T, Error>,
+) -> Result<T, Error> {
     let exit_status = child.wait()?;
-    if !exit_status.success() {
-        Err(read_git_error(stderr).unwrap_or_else(|| {
-            Error::from_git_exit_code(command, root_path.as_ref(), exit_status.code())
-        }))
-    } else {
-        Ok(())
+    if exit_status.success() {
+        if let Ok(result) = parse_result {
+            return Ok(result);
+        }
```

Comment:
Removes some indentation
```suggestion
    if exit_status.success() && parse_result.is_ok() {
        return parse_result;
```